client/engine/engine.go

```diff
@@ -358,92 +360,89 @@ func (e *Engine) handleChainEvent(chainEvent chainservice.Event) (EngineEvent, e
 	return e.attemptProgress(updatedEventHandler)
 }
 
-// handleAPIEvent handles an API Event (triggered by a client API call).
-// It will attempt to perform all of the following:
-//   - Spawn a new, approved objective (if not null)
-//   - Reject an existing objective (if not null)
-//   - Approve an existing objective (if not null)
-func (e *Engine) handleAPIEvent(apiEvent APIEvent) (EngineEvent, error) {
+// handleObjectiveRequest handles an ObjectiveRequest (triggered by a client API call).
+// It will attempt to spawn a new, approved objective.
+func (e *Engine) handleObjectiveRequest(or protocols.ObjectiveRequest) (EngineEvent, error) {
 	defer e.metrics.RecordFunctionDuration()()
 
-	if apiEvent.ObjectiveToSpawn != nil {
-
-		switch request := (apiEvent.ObjectiveToSpawn).(type) {
-
-		case virtualfund.ObjectiveRequest:
-			e.metrics.RecordObjectiveStarted(request.Id(*e.store.GetAddress()))
-			vfo, err := virtualfund.NewObjective(request, true, *e.store.GetAddress(), e.store.GetConsensusChannel)
-			if err != nil {
-				return EngineEvent{}, fmt.Errorf("handleAPIEvent: Could not create objective for %+v: %w", request, err)
-			}
-
-			err = e.registerPaymentChannel(vfo)
-			if err != nil {
-				return EngineEvent{}, fmt.Errorf("could not register channel with payment/receipt manager: %w", err)
-			}
-			return e.attemptProgress(&vfo)
-
-		case virtualdefund.ObjectiveRequest:
-			e.metrics.RecordObjectiveStarted(request.Id(*e.store.GetAddress()))
-			vdfo, err := virtualdefund.NewObjective(request, true, *e.store.GetAddress(), e.store.GetChannelById, e.store.GetConsensusChannel)
-			if err != nil {
-				return EngineEvent{}, fmt.Errorf("handleAPIEvent: Could not create objective for %+v: %w", request, err)
-			}
-			return e.attemptProgress(&vdfo)
-
-		case directfund.ObjectiveRequest:
-			e.metrics.RecordObjectiveStarted(request.Id(*e.store.GetAddress()))
-			dfo, err := directfund.NewObjective(request, true, *e.store.GetAddress(), e.store.GetChannelsByParticipant, e.store.GetConsensusChannel)
-			if err != nil {
-				return EngineEvent{}, fmt.Errorf("handleAPIEvent: Could not create objective for %+v: %w", request, err)
-			}
-			return e.attemptProgress(&dfo)
+	if or == nil {
```

Comment:
I would be in favour of just removing this if we don't expect it to happen. If `or` is `nil` the standard nil pointer error would still be enough to figure out the problem.